### PR TITLE
feat(MenuItem): separate reusable MenuItem

### PR DESCRIPTION
**Pull request by LucidityDesign**
@filiptammergard just an idea how to structure the menu. What do you think?

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -60,9 +60,9 @@ export { Stack } from "./components/layout/Stack/Stack"
 export { VerticalLayout } from "./components/layout/VerticalLayout/VerticalLayout"
 export { VerticalSpacing } from "./components/layout/VerticalSpacing/VerticalSpacing"
 export { Menu } from "./components/menus/Menu/Menu"
-export { MenuContent } from "./components/menus/Menu/MenuContent"
-export { MenuItem } from "./components/menus/Menu/MenuItem"
-export { MenuTrigger } from "./components/menus/Menu/MenuTrigger"
+// export { MenuContent } from "./components/menus/Menu/Content"
+// export { MenuItem } from "./components/menus/Menu/Item"
+// export { MenuTrigger } from "./components/menus/Menu/MenuTrigger"
```
Comment (filiptammergard) [inline src/main.ts:65]: `export { Menu } from "./components/menus/Menu/Menu"` is only exporting the Root component, right?

**Review by filiptammergard — COMMENTED** [review]
I like import/export pattern! We should probably move to that 🙂 But since this is breaking current imports for consumers we need to figure out a way to do this without breaking anything (and do the breaking in next major)

File: src/main.ts (hunk 1 above)
Comment (LucidityDesign) [reply src/main.ts:65]: Yes, I didn't consider the main yet. That's still WIP :)

File: src/components/controls/selects/SearchSelect/SearchSelect.tsx [hunk 2]
```diff
@@ -250,7 +250,7 @@ const OptionsWrapper = styled.div`
   background: ${({ theme }) => theme.colors.background.secondaryElevated};
   border-radius: ${({ theme }) => theme.borderRadii.sm};
   margin-block-start: ${({ theme }) => theme.spacer}px;
-  max-block-size: ${({ theme }) => 27 * theme.spacingBase}rem;
+  max-block-size: ${({ theme }) => 30 * theme.spacingBase}rem;
```
Comment (filiptammergard) [inline src/components/controls/selects/SearchSelect/SearchSelect.tsx:253]: What did this fix? 🙂

File: src/components/menus/MenuItem/MenuItem.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  padding-block: ${({ theme }) => 1.25 * theme.spacer}px;
+  padding-inline: ${({ theme }) => 1 * theme.spacer}px;
+  color: ${({ theme }) => theme.colors.content.primary};
+  background: ${({ theme }) => theme.colors.background.secondaryElevated};
+  border-radius: ${({ theme }) => theme.borderRadii.sm};
+  pointer-events: auto;
```
Comment (filiptammergard) [inline src/components/menus/MenuItem/MenuItem.tsx:37]: Isn't this default?

**Review by filiptammergard — COMMENTED** [review]
Am I correct that no changes in styling/functionality were done in Menu, MenuContent, MenuItem and MenuTrigger?

File: src/components/controls/selects/SearchSelect/SearchSelect.tsx (hunk 2 above)
Comment (LucidityDesign) [reply src/components/controls/selects/SearchSelect/SearchSelect.tsx:253]: Previously each item was ~43px now they are 48px. I had to adapt the container size so the user gets an indication that it's scrollable but the items are not truncated

**Comment by LucidityDesign** [comment]
> Am I correct that no changes in styling/functionality were done in Menu, MenuContent, MenuItem and MenuTrigger?

the MenuItem got a min-height of 48px, but that's it

File: src/components/menus/Menu/MenuContent.tsx [hunk 4]
```diff
@@ -1,58 +1,11 @@
...
-
-  @media ${({ theme }) => theme.mediaQueries.md} {
-    width: ${({ theme }) => 28 * theme.spacingBase}rem;
-  }
-`
+import { Content } from "./Content"
+/**
+ * @deprecated since v7.14.0. Import as `import * as Menu from "@einride/ui` and use `<Menu.Content>` instead.
+ * ```
+ * import * as Menu from "./index"
```
Comment (filiptammergard) [inline src/components/menus/Menu/MenuContent.tsx:5]: Shouldn't it be importing from `@einride/ui` here?

```suggestion
 * import * as Menu from "@einride/ui"
```

File: src/components/menus/Menu/MenuItem.tsx [hunk 5]
```diff
@@ -1,52 +1,11 @@
...
-  &:hover {
-    background: ${({ theme }) => theme.colors.background.tertiary};
-    box-shadow: none;
-  }
-`
+import { Item } from "./Item"
+/**
+ * @deprecated since v7.14.0. Import as `import * as Menu from "@einride/ui` and use `<Menu.Item>` instead.
+ * ```
+ * import * as Menu from "./index"
```
Comment (filiptammergard) [inline src/components/menus/Menu/MenuItem.tsx:5]: ?

```suggestion
 * import * as Menu from "@einride/ui"
```

File: src/components/menus/Menu/MenuTrigger.tsx [hunk 6]
```diff
@@ -1,22 +1,11 @@
...
-)
-
-const StyledTrigger = styled(DropdownMenu.Trigger)`
-  display: inline-flex;
-`
+import { Trigger } from "./Trigger"
+/**
+ * @deprecated since v7.14.0. Import as `import * as Menu from "@einride/ui` and use `<Menu.Trigger>` instead.
+ * ```
+ * import * as Menu from "./index"
```
Comment (filiptammergard) [inline src/components/menus/Menu/MenuTrigger.tsx:5]:
```suggestion
 * import * as Menu from "@einride/ui"
```

**Review by filiptammergard — APPROVED** [review]

**Pull request merged**